graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchFilterDeprecationWarningsInterceptor.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.storage.elasticsearch7;
+
+import org.graylog.shaded.elasticsearch7.org.apache.http.Header;
+import org.graylog.shaded.elasticsearch7.org.apache.http.HttpException;
+import org.graylog.shaded.elasticsearch7.org.apache.http.HttpResponse;
+import org.graylog.shaded.elasticsearch7.org.apache.http.HttpResponseInterceptor;
+import org.graylog.shaded.elasticsearch7.org.apache.http.protocol.HttpContext;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ElasticsearchFilterDeprecationWarningsInterceptor implements HttpResponseInterceptor {
+    @Override
+    public void process(HttpResponse response, HttpContext context) throws
+            HttpException, IOException {
+        List<Header> warnings = Arrays.stream(response.getHeaders("Warning")).filter(header -> !header.getValue().contains("setting was deprecated in Elasticsearch")).collect(Collectors.toList());
```

Comment:
There are some more deprecation warnings that are probably not catched by this, that we may or may not want to mute. One example is this: https://github.com/Graylog2/graylog2-server/issues/9822 (with a helpful link to the [source]( https://github.com/elastic/elasticsearch/blob/1a05a5ac24222d3e00a7f1679b12e700004af534/server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/TransportGetAliasesAction.java#L129))

Reply:
More warnings have been added to the filter